libraries/core_libs/network/src/tarcap/taraxa_capability.cpp

```diff
@@ -426,16 +426,16 @@ size_t TaraxaCapability::getReceivedBlocksCount() const { return test_state_->ge
 size_t TaraxaCapability::getReceivedTransactionsCount() const { return test_state_->getTransactionsSize(); }
 
 // PBFT
-void TaraxaCapability::sendPbftBlock(dev::p2p::NodeID const &id, PbftBlock const &pbft_block,
+void TaraxaCapability::sendPbftBlock(const dev::p2p::NodeID &id, const PbftBlock &pbft_block,
                                      uint64_t pbft_chain_size) {
   std::static_pointer_cast<PbftBlockPacketHandler>(
       packets_handlers_->getSpecificHandler(SubprotocolPacketType::PbftBlockPacket))
       ->sendPbftBlock(id, pbft_block, pbft_chain_size);
 }
 
-void TaraxaCapability::sendPbftVote(dev::p2p::NodeID const &id, std::shared_ptr<Vote> const &vote) {
-  std::static_pointer_cast<VotePacketHandler>(packets_handlers_->getSpecificHandler(SubprotocolPacketType::VotePacket))
-      ->sendPbftVote(id, vote);
+const std::shared_ptr<taraxa::network::tarcap::PacketHandler> &TaraxaCapability::getSpecificHandler(
+    SubprotocolPacketType packet_type) {
+  return packets_handlers_->getSpecificHandler(packet_type);
 }
```

Comment:
if you introduce `getSpecificHandler` why didn't you change also `sendPbftBlock` ?

Reply:
or even better would be remove it from here and keep it just in network class

Reply:
I have created anther issue https://github.com/Taraxa-project/taraxa-node/issues/1722 to get rid of all test methods in Network

Reply:
Looks like we cannot simple remove the TaraxaCapability::getSpecificHandler here and instead of keep it just in Network class. The taraxa::network::tarcap::PacketsHandler is an incomplete type std::shared_ptr<taraxa::network::tarcap::PacketsHandler>::element_type

Reply:
not sure what is going on in this discussion but you have to have `getSpecificHandler` method in both Network class as well as `TaraxaCapability` class